poppy/fresnel.py

```diff
@@ -713,17 +720,19 @@ def _propagate_stw(self, dz):
         # dz = z2 - self.z
         _log.debug("Spherical to Waist propagation, dz=" + str(dz))
 
-        if dz == 0:
+        if dz == 0*u.meter:
             _log.error("Spherical to Waist propagation stopped, no change in distance.")
             return
 
-        if dz > 0:
+        if dz > 0*u.meter:
             self._fft()
         else:
             self._inv_fft()
 
         # update to new pixel scale before applying curvature
-        self.pixelscale = self.wavelength * np.abs(dz.value) / (self.n * self.pixelscale)
+        #FIXME with units:
+        old_pixelscale = self.pixelscale
```

Comment:
I see a FIXME so I assume `old_pixelscale` will be eliminated.. just in case you missed it, it's not used anywhere